tests/extension/oneapi_non_uniform_groups/group_barrier.h

```diff
@@ -0,0 +1,215 @@
+
+  // Check the maximum number elements of type "el_type" that can be
+  // placed in the device's global and local memory. Since the test
+  // tries to allocate local and global buffers with a size equal to
+  // the work group size, the latter must be limited by the allowed
+  // buffer size.
+  uint64_t global_mem_size_in_bytes =
+      device.get_info<sycl::info::device::max_mem_alloc_size>();
+  uint64_t global_mem_size_in_elements =
+      global_mem_size_in_bytes / sizeof(el_type);
+
+  uint64_t local_mem_size_in_bytes =
+      device.get_info<sycl::info::device::local_mem_size>();
+  uint64_t local_mem_size_in_elements =
+      local_mem_size_in_bytes / sizeof(el_type);
+
+  uint64_t work_items_limit =
+      std::min(global_mem_size_in_elements, local_mem_size_in_elements);
+
+  sycl::range<1> work_group_range =
+      sycl_cts::util::work_group_range<1>(queue, work_items_limit);
+  size_t work_group_size = work_group_range.size();
+
+  for (size_t test_case = 0;
+       test_case < NonUniformGroupHelper<GroupT>::num_test_cases; ++test_case) {
+    const std::string test_case_name =
+        NonUniformGroupHelper<GroupT>::get_test_case_name(test_case);
+    INFO("Running test case (" + std::to_string(test_case) + ") with " +
+         test_case_name);
+
+    std::vector<int32_t> v(work_group_size, 0);
+    sycl::buffer<int32_t, 1> global_mem(v.data(),
+                                        sycl::range<1>(work_group_size));
+
+    sycl::buffer<sms, 1> non_uniform_group_barriers_buf(
+        non_uniform_group_barriers.data(), sycl::range<1>(5));
+
+    queue.submit([&](sycl::handler& cgh) {
+      sycl::nd_range<1> executionRange(work_group_range, work_group_range);
+
+      auto non_uniform_group_barriers_acc =
+          non_uniform_group_barriers_buf
+              .get_access<sycl::access::mode::read_write>(cgh);
+
+      sycl::local_accessor<int32_t, 1> local_acc(
+          sycl::range<1>(work_group_size), cgh);
+      sycl::accessor<int32_t, 1> global_acc =
+          global_mem.get_access<sycl::access::mode::read_write>(cgh);
+
+      cgh.parallel_for<non_uniform_group_barrier_kernel<GroupT>>(
+          executionRange, [=](sycl::nd_item<1> item) {
+            sycl::sub_group sub_group = item.get_sub_group();
+
+            // If this item is not participating in the group, leave early.
+            if (!NonUniformGroupHelper<GroupT>::should_participate(sub_group,
+                                                                   test_case))
+              return;
+
+            GroupT non_uniform_group =
+                NonUniformGroupHelper<GroupT>::create(sub_group, test_case);
+
+            size_t llid = non_uniform_group.get_local_linear_id();
+            size_t max_id = non_uniform_group.get_local_linear_range() - 1;
+
+            ASSERT_RETURN_TYPE(
+                void, sycl::group_barrier(non_uniform_group),
+                "Return type of group_barrier(GroupT g) is wrong\n");
```

Comment:
In the other test you've used plain `static_assert`. Should we unify? 
Note: my preference would be in using plain `static_assert`, because it is easier to understand at a very first glance.

Reply:
I agree. All uses of `ASSERT_RETURN_TYPE` have been made `static_assert` instead.